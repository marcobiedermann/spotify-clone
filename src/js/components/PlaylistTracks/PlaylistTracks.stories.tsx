import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import PlaylistTracks from '.';
import playlistTrackFixtures from '../PlaylistTrack/__fixtures__';

const meta: Meta<typeof PlaylistTracks> = {
  component: PlaylistTracks,
  args: {
    items: [playlistTrackFixtures],
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof PlaylistTracks>;

const Default: Story = {};

export { Default };
export default meta;
