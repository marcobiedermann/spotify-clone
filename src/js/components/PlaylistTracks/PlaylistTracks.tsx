import PlaylistTrack, { PlaylistTrackProps } from '../PlaylistTrack';

export interface PlaylistTracksProps {
  items: PlaylistTrackProps[];
}

function PlaylistTracks(props: PlaylistTracksProps): JSX.Element {
  const { items = [] } = props;

  return (
    <table>
      <thead>
        <tr>
          <th>#</th>
          <th>Title</th>
          <th>Album</th>
          <th>Added</th>
          <th align="right">Duration</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <PlaylistTrack key={item.track?.id} {...item} />
        ))}
      </tbody>
    </table>
  );
}

export default PlaylistTracks;
