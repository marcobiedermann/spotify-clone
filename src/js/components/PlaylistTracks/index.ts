import PlaylistTracks from './PlaylistTracks';

export default PlaylistTracks;
export * from './PlaylistTracks';
