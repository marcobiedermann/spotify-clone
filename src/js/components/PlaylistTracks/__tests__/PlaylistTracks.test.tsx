import { render } from '@testing-library/react';
import React from 'react';
import PlaylistTracks from '..';

it('renders correctly', () => {
  const { container } = render(<PlaylistTracks />);

  expect(container.firstChild).toMatchSnapshot();
});
