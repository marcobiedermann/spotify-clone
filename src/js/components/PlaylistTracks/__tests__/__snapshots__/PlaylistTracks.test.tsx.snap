// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Title
      </th>
      <th>
        Album
      </th>
      <th>
        Added
      </th>
      <th
        align="right"
      >
        Duration
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
